mlir/lib/Dialect/SPIRV/IR/SPIRVTypes.cpp

```diff
@@ -532,7 +538,7 @@ void ScalarType::getExtensions(SPIRVType::ExtensionArrayRefVector &extensions,
     [[fallthrough]];
   case StorageClass::Input:
   case StorageClass::Output:
-    if (getIntOrFloatBitWidth() == 16) {
+    if (getIntOrFloatBitWidth() == 16 && !llvm::isa<BFloat16Type>(*this)) {
```

Comment:
nit: no need to prefix cast functions with `llvm::`. The existing code was automatically converted to the new casting style via clang tidy that added this namespace prefixes.